crates/matrix-sdk/src/widget/mod.rs

```diff
@@ -269,16 +291,42 @@ impl WidgetDriver {
 
                             Some(event) = matrix.recv() => {
                                 // Forward all events to the incoming messages stream.
-                                let _ = incoming_msg_tx.send(IncomingMessage::MatrixEventReceived(event));
+                                let _ = incoming_msg_tx.send(IncomingMessage::MatrixEventReceived(MatrixEvent::Timeline(event)));
                             }
                         }
                     }
                 });
             }
-
-            Action::Unsubscribe => {
+            Action::UnsubscribeTimeline => {
                 self.event_forwarding_guard = None;
             }
+            Action::SubscribeToDevice => {
+                // Only subscribe if we are not already subscribed.
+                if self.to_device_event_forwarding_guard.is_none() {
+                    let (stop_forwarding, guard) = {
+                        let token = CancellationToken::new();
+                        (token.child_token(), token.drop_guard())
+                    };
+
+                    self.to_device_event_forwarding_guard = Some(guard);
+                    let (mut matrix, events_tx) =
+                        (matrix_driver.to_device_events(), incoming_msg_tx.clone());
+                    tokio::spawn(async move {
```

Comment:
Please use `matrix_sdk_common::executor::spawn` instead.

Reply:
We need to be careful here for wasm since it will drop the handle and consequently stop the async process.
So this would need a `forget` call on the handle or keeping a reference to the handle while the process should be running (tokio and the wasm spawn behave different in that regard)

Reply:
So please update `matrix_sdk_common::executor::spawn` to do what you need, but don't use `tokio::spawn` directly here.